src/main/java/com/blackducksoftware/integration/hub/alert/channel/hipchat/HipChatChannel.java

```diff
@@ -73,6 +81,41 @@ public String getApiUrl() {
         return HIP_CHAT_API;
     }
 
+    @Override
+    public String testGlobalConfig(final GlobalHipChatConfigEntity entity) {
+        if (entity == null) {
+            return "The provided entity was null.";
+        }
+        final RestConnection restConnection = channelRestConnectionFactory.createUnauthenticatedRestConnection(HIP_CHAT_API);
```

Comment:
For tests methods like this we can create our own event with appropriate ProjectData passed in and our test entity. Then pass those values to the original rest connection in the parent class and have everything done automatically.

Reply:
That breaks the paradigm. Events may contain project data and configuration info, but they aren't meant to be a convenience class (aka a wrapper). Events should only ever be created when we are processing notifications.